lib/gooddata/rest/connection.rb

```diff
@@ -382,6 +383,47 @@ def get(uri, options = {}, &user_block)
         request(:get, uri, nil, options, &user_block)
       end
 
+      def stream(uri, filename)
+        uri = URI.join(server_url, uri.to_s)
+        puts "Streaming URL #{uri.to_s}"
+
+        q = { headers: @webdav_headers.merge(:x_gdc_authtt => headers[:x_gdc_authtt]) }
+        File.open(filename, 'wb+') do |fout|
+          Net::HTTP.start(uri.host, uri.port,
+                          q.merge(:use_ssl => uri.scheme == 'https',
+                                  :verify_mode => @verify_ssl)) do |http|
+            request = Net::HTTP::Get.new(uri)
+            request.initialize_http_header({'Accept'=> 'text/csv'})
+            q[:headers].each do |k,v|
+              request[k] = v
+            end
+            request.basic_auth(@server.options[:username], @server.options[:password])
+
+            req_ok = false
+            retries = 0
+            loop do
+              http.request(request.dup) do |response|
+                if response.code.to_i == 200 then
```

Comment:
This will retry the request even if we get something else than 202. For example, it is not sensible to repeat the request when we get 400.